app/state.go

```diff
@@ -5,16 +5,18 @@ import merkle "github.com/tendermint/go-merkle"
 // State represents the app states, separating the commited state (for queries)
 // from the working state (for CheckTx and AppendTx)
 type State struct {
-	committed merkle.Tree
-	appendTx  merkle.Tree
-	checkTx   merkle.Tree
+	committed  merkle.Tree
+	appendTx   merkle.Tree
+	checkTx    merkle.Tree
+	persistent bool
 }
 
-func NewState(tree merkle.Tree) State {
+func NewState(tree merkle.Tree, persistent bool) State {
```

Comment:
It would be cleaner if persistence could be attained could be attained directly from the tree. see: tendermint/go-merkle#18